### PR TITLE
feat: support undefined separate from null

### PR DESCRIPTION
**Pull request by helios1138**

### CODE REVIEW

File: src/Decoder.ts [hunk 1]
```diff
@@ -44,7 +44,9 @@ try {
   EMPTY_VIEW.getInt8(0);
 } catch (e) {
   if (!(e instanceof RangeError)) {
-    throw new Error("This module is not supported in the current JavaScript engine because DataView does not throw RangeError on out-of-bounds access");
+    throw new Error(
+      "This module is not supported in the current JavaScript engine because DataView does not throw RangeError on out-of-bounds access",
+    );
```
Comment (bfelbo) [inline src/Decoder.ts]: Can you disable and revert any IDE linting changes? That'll make it easier to merge any upstream changes into this repo :)

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@msgpack/msgpack",
-  "version": "2.8.0",
+  "name": "msgpack",
+  "version": "1.0.0",
   "description": "MessagePack for ECMA-262/JavaScript/TypeScript",
   "author": "The MessagePack community",
```
Comment (bfelbo) [inline package.json]: Let's also update the other keys in `package.json`, e.g. `description` and `repository`. Let's make sure to mention that it's a fork in the description.

File: src/Decoder.ts (hunk 1 above)
Comment (bfelbo) [reply src/Decoder.ts]: (see also extra line break below)

File: test/edge-cases.test.ts [hunk 3]
```diff
@@ -71,17 +71,17 @@ describe("edge cases", () => {
...
-      assert.throws(() => {
-        decoder.decode([TYPE_NEVER_USED]);
-      }, /unrecognized type byte/i);
-      testDecoder(decoder);
-    });
-  });
+  // context("try to decode invalid MessagePack binary", () => {
+  //   it("throws errors", () => {
+  //     const decoder = new Decoder();
+  //     const TYPE_NEVER_USED = 0xc1;
```
Comment (bfelbo) [inline test/edge-cases.test.ts]: Does this work if you just change it to `0xc4`? If so, that'd be cleaner.

Otherwise, I'm good with commenting it out.

**Review by bfelbo — APPROVED** [review]
LGTM!

We could consider moving the `package.json` changes into a separate PR.

File: test/edge-cases.test.ts (hunk 3 above)
Comment (helios1138) [reply test/edge-cases.test.ts]: No, once we add 0xc1, this closes the entire usable range from 0xc0 to 0xe0 (I checked every one 🙂)

Comment (bfelbo) [reply test/edge-cases.test.ts]: Makes sense. Should we do `it.skip(...)` to make it clear that the test is skipped and add a comment why?

File: test/edge-cases.test.ts [hunk 4]
```diff
@@ -71,7 +71,8 @@ describe("edge cases", () => {
     });
   });
 
-  context("try to decode invalid MessagePack binary", () => {
+  // No longer relevant because 0xc1 is used to encode `undefined`
+  context.skip("try to decode invalid MessagePack binary", () => {
```
Comment (mieszko4) [inline test/edge-cases.test.ts:75]: I guess you could enable test with another unused hexa

Comment (helios1138) [reply test/edge-cases.test.ts:75]: Entire allowed range is used after this change, so it doesn't look possible
